featuregate/doc.go

```diff
@@ -0,0 +1,25 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+// Package featuregate provides mechanisms for feature flagging in the OpenTelemetry Collector.
+//
+// Example (Checking if a feature gate is enabled):
+//
+//	gate := featuregate.GlobalRegistry().MustRegister(
```

Comment:
The testable examples below will already appear in godoc. No need to duplicate things here, especially since it's never executed and will easily be stale.